File: util/src/main/java/fr/edjaz/util/http/HttpErrorInfo.java
package fr.edjaz.util.http;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.time.ZonedDateTime;

@Getter
public class HttpErrorInfo {
    private final ZonedDateTime timestamp;
    private final String path;
    private final HttpStatus httpStatus;
    private final String message;


    public HttpErrorInfo() {
        timestamp = null;
        this.httpStatus = null;
        this.path = null;
        this.message = null;
    }

    public HttpErrorInfo(HttpStatus httpStatus, String path, String message) {
        timestamp = ZonedDateTime.now();
        this.httpStatus = httpStatus;
        this.path = path;
        this.message = message;
    }

    public String getError() {
        return httpStatus.getReasonPhrase();
    }

}

File: util/src/main/kotlin/fr/edjaz/util/http/HttpErrorInfo.kt
package fr.edjaz.util.http

import org.springframework.http.HttpStatus
import java.time.ZonedDateTime

class HttpErrorInfo {
    var timestamp: ZonedDateTime?
    var path: String?
    var httpStatus: HttpStatus?
    var message: String?

    constructor() {
        timestamp = ZonedDateTime.now()
        this.httpStatus = null
        this.path = null
        this.message = null
    }

    constructor(httpStatus: HttpStatus?, path: String?, message: String?) {
        timestamp = ZonedDateTime.now()
        this.httpStatus = httpStatus
        this.path = path
        this.message = message
    }

    fun getError(): String = httpStatus!!.reasonPhrase
}
